frontend/src/modules/Shared/Shares/ShareEditForm.js

```diff
@@ -173,7 +192,7 @@ const ItemRow = (props) => {
           )}
           {possibleAction === 'Nothing' && (
             <Typography color="textSecondary" variant="subtitle2">
-              Wait until this item is processed
+              Wait until this item is processed and/or re-apply task is complete
```

Comment:
Are we enforcing this in the backend?

Reply:
addressed in latest commit and tested again to ensure revoke still works for healthy share item and throws error for PendingReApply item

For now only enforcing this check on revoke items - the more closer `healthstatus` and `status` get related - the more I am tempted to include transitions and such in the already existing state machine... for now I am leaving as is to not increase complexity / scope of this PR